src/main/scala/ai/privado/javascript/passes/methodfullname/MethodFullName.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * This file is part of Privado OSS.
+ *
+ * Privado is an open source static code analysis tool to discover data flows in the code.
+ * Copyright (C) 2022 Privado, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, contact support@privado.ai
+ */
+
+package ai.privado.javascript.passes.methodfullname
+
+import ai.privado.semantic.Language.finder
+import io.shiftleft.codepropertygraph.generated.{Cpg, Operators}
+import io.shiftleft.codepropertygraph.generated.nodes.Call
+import io.shiftleft.codepropertygraph.generated.nodes.Call.PropertyNames
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import io.shiftleft.semanticcpg.language._
+import overflowdb.traversal.Traversal
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+class MethodFullName(cpg: Cpg) extends ConcurrentWriterCpgPass[(String, String, String, String)](cpg) {
+
+  val cachedCall         = cpg.call.whereNot(_.name(Operators.ALL.asScala.toSeq: _*)).l
+  val cachedOperatorCall = cpg.call(Operators.assignment).l
+  override def generateParts(): Array[(String, String, String, String)] = {
+
+    val requireStyleDependency = cpg.dependency.name
+      .flatMap(dependencyName => {
+        Traversal(cachedOperatorCall)
+          .where(
+            _.argument(2)
+              .code(".*require.*('" + dependencyName + "'|\"" + dependencyName + "\").*")
+          )
+          .argument(1)
+          .isIdentifier
+          .map(item => (item.name, dependencyName, item.location.filename, "pkg."))
+      })
+      .toArray
+
+    val importStyleDependency = cpg.staticImport
+      .map(staticImport =>
+        (
+          staticImport.importedAs.getOrElse(""),
+          staticImport.importedEntity.getOrElse(""),
+          staticImport.file.head.name,
+          "pkg."
+        )
+      )
+      .toArray
+
+    val consoleNodes = cpg.file.map(fileName => ("console", "console", fileName.name, "")).toArray
+
+    importStyleDependency ++ requireStyleDependency ++ consoleNodes
+  }
+
+  override def runOnPart(builder: DiffGraphBuilder, importedTuple: (String, String, String, String)): Unit = {
```

Comment:
@khemrajrathore Add comment for this method as well as specific queries that you are firing.